internal/p2p/pqueue_test.go

```diff
@@ -0,0 +1,45 @@
+package p2p
+
+import (
+	"testing"
+	"time"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+func TestCloseWhileDequeueFull(t *testing.T) {
+	enqueueLength := 5
+	chDescs := []ChannelDescriptor{
+		{ID: 0x01, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x02, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x03, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x04, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x05, Priority: 1, MaxSendBytes: 4},
+		{ID: 0x06, Priority: 1, MaxSendBytes: 4},
+	}
+	pqueue := newPQScheduler(log.NewNopLogger(), NopMetrics(), chDescs, uint(enqueueLength), 1, 120)
+
+	for i := 0; i < enqueueLength; i++ {
+		pqueue.enqueue() <- Envelope{
+			channelID: 0x01,
+			Message:   &testMessage{Value: "foo"}, // 5 bytes
+		}
+	}
+
+	go pqueue.process()
+
+	// sleep to allow context switch for process() to run
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Wish there was a cleaner way to know process() is executing. Perhaps we could add a status field or something to the queue. Not really a blocker, just wanting to point it out.